### PR TITLE
Implement `DictManager`

### LINKED ISSUE #179: Implement DictManager
[Reference](https://github.com/lambdaclass/cairo-vm/blob/main/vm/src/hint_processor/builtin_hint_processor/dict_manager.rs)

### PR DESCRIPTION
**Pull request by fmoletta**

### CODE REVIEW

File: pkg/hints/dict_manager/dict_manager.go [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+}
+
+func NewDefaultDictionary(defaultValue *MaybeRelocatable, dict *map[MaybeRelocatable]MaybeRelocatable) Dictionary {
+	return Dictionary{
+		dict:         *dict,
+		defaultValue: defaultValue,
+	}
+}
+
+func NewDictionay(dict *map[MaybeRelocatable]MaybeRelocatable) Dictionary {
```
Comment (toni-calvin) [inline pkg/hints/dict_manager/dict_manager.go]:
```suggestion
func NewDictionary(dict *map[MaybeRelocatable]MaybeRelocatable) Dictionary {
```

File: pkg/hints/dict_manager/dict_manager.go [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+		data:       NewDefaultDictionary(defaultValue, dict),
+		currentPtr: base,
+	}
+}
+
+func (d *DictTracker) CopyDictionary() map[MaybeRelocatable]MaybeRelocatable {
+	return d.data.dict
+}
+
+func (d *DictTracker) GetValue(key *MaybeRelocatable) *MaybeRelocatable {
```
Comment (toni-calvin) [inline pkg/hints/dict_manager/dict_manager.go]: Question: Is it safe to return nil here and not manage an error?

**Review by toni-calvin — APPROVED** [review]
LGTM, I added a typo comment and a question 😄

File: pkg/hints/dict_manager/dict_manager.go (hunk 1 above)
Comment (fmoletta) [reply pkg/hints/dict_manager/dict_manager.go]: Ty

File: pkg/hints/dict_manager/dict_manager.go (hunk 2 above)
Comment (fmoletta) [reply pkg/hints/dict_manager/dict_manager.go]: Good call

Comment (fmoletta) [reply pkg/hints/dict_manager/dict_manager.go]: Changed return to (*MaybeRelocatable, error)

**Review by jrchatruc — APPROVED** [review]

**Pull request merged**